### ISSUE TITLE
Crashlytics with Swift Package Manager run script not working properly

### ISSUE THREAD

**Issue by milan-prijic-lu** [firebase/firebase-ios-sdk#7634]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 12.3
  * Firebase SDK version: 7.7.0
  * Installation method: Swift Package Manager
  * Firebase Component: Crashlytics

#### Steps to reproduce:

We decided to migrate from using Carthage to SPM. We used documentation here to setup the project: https://github.com/firebase/firebase-ios-sdk/blob/master/SwiftPackageManager.md
We are using Analytics and Crashlytics from Firebase. Analytics works perfectly without any issues. Crashlytics on the other hand do not send crash reports to our dashboard ever! Here is the list of what steps we already did:

1) We added -Objc linker flag into build settings
2) We added DWARF with dSYMs in build settings
3) We added run script for crashlytics: 
echo "Script started"
"${BUILD_DIR%Build/*}/SourcePackages/checkouts/firebase-ios-sdk/Crashlytics/run"

Even though this script is stated in your documentation for Crashlytics it does not work! Here is what happens during build time:
Script started
Running upload-symbols in Build Phase mode
Validating build environment for Crashlytics...
Validation succeeded. Exiting because upload-symbols was run in validation mode

4) After that we found some post on your github about running upload-symbols file in script other than run, so we changed the script to: 
echo "Script started"
"${BUILD_DIR%Build/*}/SourcePackages/checkouts/firebase-ios-sdk/Crashlytics/upload-symbols" --build-phase

Now that works, and when building it logs:
Script started
Running upload-symbols in Build Phase mode
Validating build environment for Crashlytics...
Processing dSYMs...
Already uploaded this dSYM, not doing it again (this line previously stated the uploaded file, so all good on that)

The problem is that even with this success, 0 crashes that we produce are visible on Crashlytics dashboard! (we are making the app crash both with fatalError() and by producing nil values and unwrapping them - same results...). We removed all filters on our Crashlytics dashboard, still no results. Also we are trying to make this work for 3 days now, so enough time has passed from sending the reports until they should be visible on dashborad.

**Comment by elenadoty** [comment]
Hi @milan-prijic-lu - are you seeing any missing dSYM banners on your dashboard, or just no crashes?

**Comment by milan-prijic-lu** [comment]
No, we checked that one too, nothing at all (and we send like at least 20-30 crashes during last 2-3 days)

**Comment by Appswage** [comment]
I will say that I just integrated it with the same steps as you. However, it is recognizing crashes. What it isn't doing is sending the dSYM's as I get the missing banner in Crashlytics.

**Comment by milan-prijic-lu** [comment]
Not sure why is that happening, here are the screenshots from our dashboard:

![Screenshot 2021-03-03 at 10 59 11](https://user-images.githubusercontent.com/43952537/109788424-99b40e80-7c0f-11eb-939a-b2cb21f5e7ba.png)
![Screenshot 2021-03-03 at 10 59 07](https://user-images.githubusercontent.com/43952537/109788429-9ae53b80-7c0f-11eb-9cbb-965d291e63b4.png)

Could it be something buggy from changing the project from Carthage to SPM?
Are you using fatalError() to crash the app?
Simulator or real device?

I am just trying to find out why we are not even seeing missing dSYM's on our dashboard...it makes no sense...

**Comment by Appswage** [comment]
I am using a real device and did use fatalError(). A few comments I could make and sorry if they are basic. Just trying to cover the areas from my own experience.

1/ Certainly seems like it has been long enough here but sometimes it does take upwards of a day to process. For example, on my new project the first two crashes showed after about 10 minutes. I crashed it numerous times after trying to debug the missing dSYMs and those didn't show up until the next day. Seems like it has been much longer for you so likely not the issue.

2/ You cannot crash while attached to the debugger. I am sure you know this but it happens. Debugger attached crashes don't register. 

3/ Sometimes I have used fataError() from within the AppDelegate and that worked while other times it did not. Best to put it deeper into the app to ensure everything had time to spin up.

4/ Try a different Xcode project as a test to see if you have a more generalized problem or a project specific issue. 

5/ Remove SPM modules and reintegrate.

6/ Contact Firebase support. I have seen issues with the project on the site that they had to resolve. People have reported this. Worth asking their support to take a look on their side.

Hopefully something works for you soon and then you can tell me how to get the dSYM's processed without handling it separately...

**Comment by milan-prijic-lu** [comment]
Hi, first of all, thank you for help: (issue is now fixed so here is what I have found)

1) Nope, once it was fixed dashboard responded within minute or two, so it was definitely not time issue 

2) This was the problem, but not how you would expect it:
![Screenshot 2021-03-04 at 11 19 00](https://user-images.githubusercontent.com/43952537/109952483-2c6fae80-7cdf-11eb-8b95-2533b8067f0c.png)

So the build configuration needs to be "debug", and "debug executable" checked off. This is the only way it works, and it works with "${BUILD_DIR%Build/*}/SourcePackages/checkouts/firebase-ios-sdk/Crashlytics/run" (not working with upload-symbols script mentioned in the post, with upload-symbols dSYM's are not being sent to dashboard). This solution works both on simulator and on real devices.

3) I would agree on that one, from the start we have been inserting fatalError() deep inside the app

4) 5) 6) Those were not necessary since issue was fixed

Again this was helpful, and thank you! My suggestion would be to add these to setup documentation. Firebase documentation uses cocoa-pods and old Swift/Firebase versions and is miss-leading in most cases.

**Comment by paulb777** [comment]
Hmm, Crashlytics should work with a Release configuration.

**Comment by Appswage** [comment]
To be clear, are you saying that the crashes are now registering and the dSYMs are sending? I do have the "Debug executable" selected and the crashes are registering, but the DSYMs are not sending. Not clear if you have solved both issues or just the ability to get a crash logged on Crashlytics.

**Comment by milan-prijic-lu** [comment]
As soon as I checked off the "Debug executable" all my crash reports appeared on Dashboard. They were all missing DSYMs, but that because I used:
"${BUILD_DIR%Build/*}/SourcePackages/checkouts/firebase-ios-sdk/Crashlytics/upload-symbols" --build-phase

Once I changed to:
"${BUILD_DIR%Build/*}/SourcePackages/checkouts/firebase-ios-sdk/Crashlytics/run"

DSYMs were uploaded and everything worked properly. I suppose you know but just to mention, you build it with "debug" and "debug executable - unchecked", produce a crash, start the app manually from home screen of the simulator. Thats at least what I did....
